builtin/multi-pack-index.c

```diff
@@ -127,6 +135,8 @@ static int cmd_multi_pack_index_write(int argc, const char **argv)
 
 	FREE_AND_NULL(options);
 
+	normalize_object_dir();
```

Comment:
Rather than calling this in each subcommand, you can _possibly_ do it with `OPT_CALLBACK`:

```
static struct option common_opts[] = {
	OPT_CALLBACK(0, "object-dir", &opts.object_dir, N_("file"),
		     N_("object directory containing set of packfile and pack-index pairs"),
		     normalize_object_dir),
	OPT_END(),
};
```

The function signature of `normalize_object_dir` would need to change to:

```
static int normalize_object_dir(const struct option *opt, const char *arg, int unset)
```

...and you might need to copy some of the filename processing from `parse-options.c:get_value()` ([link](https://github.com/microsoft/git/blob/vfs-2.36.0/parse-options.c#L170-L181)), but it would centralize the processing and avoid needing to make sure we have `normalize_object_dir` in every new subcommand.

Reply:
Great idea! That's much cleaner. The initialization to the `.git/objects` directory can happen before parsing.

I've got this sent upstream, too. Do you want to add that suggestion there, too?